synapse/handlers/register.py

```diff
@@ -362,58 +360,6 @@ def appservice_register(self, user_localpart, as_token):
         )
         return user_id
 
-    @defer.inlineCallbacks
```

Comment:
please can the bits of this PR which kill off old dead code be factored out to a separate PR?

Reply:
Of course, sorry about that.

Reply:
https://github.com/matrix-org/synapse/pull/5963